sqlmesh/core/plan/definition.py

```diff
@@ -63,6 +63,19 @@ class Plan(PydanticModel, frozen=True):
     effective_from: t.Optional[TimeLike] = None
     execution_time: t.Optional[TimeLike] = None
 
+    user_defined_flags: t.Optional[
```

Comment:
I believe `user_provided_*` is more accurate. We define the flags, users provide them